### PR TITLE
Admin member area a11y fixes: Overview page

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR encompasses some minor fixes to ensure accessibility of the member detail view, focusing first on the Overview tab.  

_There are a couple of other places where IDs are duplicated and therefore labels don't get correctly associated, but I think these are all fixed in Nicks E2E tests PR: https://github.com/forem/forem/pull/16513. I'm going to keep this PR in draft until that's merged and I can update from it, as I think we have a few other areas of overlap too._
 
## Related Tickets & Documents

Contributes toward https://github.com/forem/forem/issues/16504

## QA Instructions, Screenshots, Recordings

- Check out the new member detail view locally - it should all look the same, with the exception of the "tags moderated", which should no longer have the 'X' icon (since a user can't delete them from this page).
- Inline with the code changes I've added some notes on what's changed and why, feel free to check those areas with a screen reader switched on, or e.g. inspect input labels/descriptions in dev tools

### UI accessibility concerns?

Addressing several small accessibility issues

## Added/updated tests?

- [ ] Yes
- [X] No, and this is why: E2E tests are currently being written separately. I'll update this branch once they're merged to main, and make sure I make any necessary changes.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [X] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Review by juliannatetreault — APPROVED** [review]
Tested locally and all works as expected! Thanks, @aitchiss! 🚀

File: app/views/admin/users/_stats.html.erb [hunk 1]
```diff
@@ -1,30 +1,32 @@
-<section class="crayons-card mb-7 px-2 py-3 flex flex-col l:flex-row gap-4 l:gap-0">
-  <div class="flex flex-1">
-    <div class="align-center flex-1">
-      <strong class="fs-xl l:fs-2xl fw-heavy lh-tight"><%= @user.comments_count %></strong>
-      <p class="fs-s l:fs-base color-secondary">Comments</p>
+<section class="crayons-card mb-7 px-4 m:px-8 py-3 ">
+  <dl class="grid grid-flow-col grid-cols-3 grid-rows-2 l:grid-cols-6 l:grid-rows-1">
```
[the remark this answers is not in the document]
Comment (nickytonline) [reply app/views/admin/users/_stats.html.erb:2]: For folks like me who were wonderinf if it was OK to wrap `<dd />`/`<dt />` pairs in the list with `<div />`s, see [Wrapping name-value groups in div elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl#wrapping_name-value_groups_in_div_elements)

File: app/views/admin/users/widgets/_tag_moderation.html.erb [hunk 2]
```diff
@@ -12,15 +12,16 @@
...
           #<%= role.resource_name.to_s %>
-          <%= crayons_icon_tag(:x, class: "c-pill__icon", width: 18, height: 18) %>
         </button>
       </li>
     <% end %>
   </ul>
 <% end %>
 
-<div class="crayons-notice crayons-notice--warning">
+<div id="tag-moderation-description" class="crayons-notice crayons-notice--warning">
```
[the remark this answers is not in the document]
Comment (nickytonline) [reply app/views/admin/users/widgets/_tag_moderation.html.erb:25]: cc: @amywtlin

File: cypress/integration/seededFlows/adminFlows/users/manageRoles.spec.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // More on roles, https://admin.forem.com/docs/forem-basics/user-roles
-function openCreditsModal() {
```
Comment (nickytonline) [inline cypress/integration/seededFlows/adminFlows/users/manageRoles.spec.js:2]: That was my copy pasta gaffe 🙃

**Review by nickytonline — APPROVED** [review]
Thanks @aitchiss 🚀

**Pull request merged**